employee-rostering-gwtui/src/main/java/org/optaweb/employeerostering/gwtui/client/viewport/shiftroster/ShiftGridObjectPopup.java

```diff
@@ -111,27 +119,52 @@ public void init(final ShiftGridObject shiftGridObject) {
         shiftGridObject.getElement().classList.add("selected");
         final ShiftView shift = shiftGridObject.getShiftView();
 
+        setup(shift);
+
+        popupFactory.getFormPopup(this).ifPresent((fp) -> {
+            formPopup = fp;
+            formPopup.showFor(shiftGridObject);
+        });
+    }
+
+    public void createNewShift() {
+        setup(new ShiftView(tenantStore.getCurrentTenantId(), new Spot(), LocalDateTime.now(), LocalDateTime.now().plusHours(9)));
+
+        spotSelect.setEnabled(true);
+        deleteButton.remove();
+        popupTitle.innerHTML = translationService.format(OptaWebUIConstants.ShiftRosterToolbar_createShift);
+
+        popupFactory.getFormPopup(this).ifPresent((fp) -> {
+            formPopup = fp;
+            formPopup.center();
+        });
+    }
+
+    private void setup(ShiftView shiftView) {
         employeeSelect.clear();
         employeeSelect.addItem(translationService.format(OptaWebUIConstants.Shift_unassigned), "-1");
 
         SpotRestServiceBuilder.getSpotList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(spots -> {
             spots.forEach(s -> this.spotSelect.addItem(s.getName(), s.getId().toString()));
-            spotSelect.setSelectedIndex(spots.indexOf(shiftGridObject.getSpot()));
+            if (null == shiftView.getSpotId()) {
+                spotSelect.setSelectedIndex(0);
+            } else {
+                spotSelect.setSelectedIndex(spots.indexOf(shiftGridObject.getSpot()));
+            }
         }));
 
         EmployeeRestServiceBuilder.getEmployeeList(tenantStore.getCurrentTenantId(), FailureShownRestCallback.onSuccess(employees -> {
             employees.forEach(e -> employeeSelect.addItem(e.getName(), e.getId().toString()));
-            employeeSelect.setSelectedIndex((shiftGridObject.getEmployee() == null) ? 0 : employees.indexOf(shiftGridObject.getEmployee()) + 1);
+            if (null == shiftView.getEmployeeId()) {
```

Comment:
No need for [Yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions) in Java.